hazelcast/src/main/java/com/hazelcast/client/impl/clientside/HazelcastClientInstanceImpl.java

```diff
@@ -860,8 +863,9 @@ public void waitForInitialMembershipEvents() {
 
     public void sendStateToCluster() throws ExecutionException, InterruptedException {
         userCodeDeploymentService.deploy(this);
-        proxyManager.createDistributedObjectsOnCluster();
+        schemaService.sendAllSchemas();
         queryCacheContext.recreateAllCaches();
+        proxyManager.createDistributedObjectsOnCluster();
```

Comment:
Probably not related to this PR but we make 4 invocations for the state transfer. I wish we had a mechanism in place to combine all into a single invocation and let the server handle the rest and server can handle them in parallel.

Reply:
I will remove sendAllSchemas. I think we have a simpler better alternative which is to clear the schemas from the client on cluster restart. For your other suggestion, I can say that combining client messages from different logics is not always easy but we can try.

Reply:
We have decided to make `sendAllSchemas` async. 
Also ClientSchemaService should do the local registration after invocation is successful.